src/Adapter/Product/AbstractProductHandler.php

```diff
@@ -129,4 +130,29 @@ protected function validateField(Product $product, string $field, int $errorCode
             );
         }
     }
+
+    /**
+     * @param Product $product
+     * @param int $errorCode
+     *
+     * @throws CannotUpdateProductException
+     * @throws ProductException
+     */
+    protected function performUpdate(Product $product, int $errorCode): void
+    {
+        try {
+            if (false === $product->update()) {
+                throw new CannotUpdateProductException(
+                    sprintf('Failed to update product #%d', $product->id),
+                    $errorCode
+                );
+            }
+        } catch (PrestaShopException $e) {
+            throw new ProductException(
```

Comment:
As i remember we used to throw generic exception in this case so it shows that it wasn't really the update that caused it, but rather some connection problems. idk really I don't mind putting `..Update..Exception`

Reply:
there is no point allowing partially set default category when we are reassigning whole category tree each time :thinking:

Reply:
Ok good for me, I was just surprised because the previous private method used the `CannotUpdateProductException` It makes sense to use a different Exception but then maybe it shouldn't be a `ProductException` either but an even more generic

Although when I check at all other handlers they use the same convention so `ProductException` is the best choice